src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -131,6 +131,7 @@ protected function warmup($warmupDir, $realCacheDir, $enableOptionalWarmers = tr
             $realKernelClass = substr($realKernelClass, $pos + 1);
         }
         $tempKernel = $this->getTempKernel($realKernel, $namespace, $realKernelClass, $warmupDir);
+        $tempKernel->declaredClasses = $realKernel->declaredClasses;
```

Comment:
What about setting `declaredClasses` private with only a getter and modifying `getTempKernel` to set it's value in the code instead of relying on a public argument?